Bluewire.Tools.GitRepository.IntegrationTests/BranchIntegrationPointLocatorTests.cs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        [Test]
+        public async Task SubjectOnMergedBranchResolvesToIdentityOfMergeCommit()
+        {
+            await builder.CreateBranchWithCommits(startTag, "branch", 4);
+            await builder.AddCommitsToBranch("master", 3);
+
+            await session.Merge(workingCopy, new Ref("branch"));
+            var mergeCommit = await session.ResolveRef(workingCopy, MasterBranch);
+
+            await builder.AddCommitsToBranch("master", 2);
+
+            var subjectRef = new Ref("branch").Ancestor(2); // On branch, but not at the tip.
+
+            var resolved = await sut.FindCommit(workingCopy, startTag, MasterBranch, subjectRef);
+
+            Assert.That(resolved, Is.EqualTo(mergeCommit));
+        }
+
+        [Test, Description("Ancestry chains match as far as they both go, but first-parent chain has been exhausted: subject was merged immediately after start.")]
+        public async Task SubjectOnBranchMergedImmediatelyAfterStartResolvesToIdentityOfMergeCommitImmediatelyFollowingStart()
+        {
+            await builder.CreateBranchWithCommits(startTag, "branch", 4);
+            await builder.AddCommitsToBranch("master", 2);
+            var thisStart = await session.ResolveRef(workingCopy, MasterBranch);
+
+            await session.Merge(workingCopy, new Ref("branch"));
+            var mergeCommit = await session.ResolveRef(workingCopy, MasterBranch);
+
+            await builder.AddCommitsToBranch("master", 2);
+
+            var subjectRef = new Ref("branch").Ancestor(2); // On branch, but not at the tip.
+
+            var resolved = await sut.FindCommit(workingCopy, thisStart, MasterBranch, subjectRef);
+
+            Assert.That(resolved, Is.EqualTo(mergeCommit));
+        }
+
+        [Test, Description("Ancestry chains match as far as they both go, but subject chain has been exhausted and first-parent chain has not: subject is the tip of the merged branch.")]
+        public async Task SubjectAtTipOfMergedBranchResolvesToIdentityOfMergeCommit()
+        {
+            await builder.CreateBranchWithCommits(startTag, "branch", 1);
+            await builder.AddCommitsToBranch("master", 2);
+            var thisStart = await session.ResolveRef(workingCopy, MasterBranch);
+
+            await session.Merge(workingCopy, new Ref("branch"));
+            var mergeCommit = await session.ResolveRef(workingCopy, MasterBranch);
+
+            await builder.AddCommitsToBranch("master", 2);
+
+            var resolved = await sut.FindCommit(workingCopy, thisStart, MasterBranch, new Ref("branch"));
+
+            Assert.That(resolved, Is.EqualTo(mergeCommit));
+        }
+
+        [Test, Description("Ancestry chains match entirely, but we know that subject != start: subject is the tip of the merged branch and was merged immediately after start.")]
+        public async Task SubjectAtTipOfBranchMergedImmediatelyAfterStartResolvesToIdentityOfMergeCommitImmediatelyFollowingStart()

```

Comment:
I think it has something to do with multiple commits capable of being 'after' the start commit, but only one can 'follow' it on the first-parent ancestry chain. Or something.